sympy/core/tests/test_subs.py

```diff
@@ -862,3 +862,13 @@ def test_issue_19558():
 
     assert e.subs(x, oo) == AccumBounds(-oo, oo)
     assert (sin(x) + cos(x)).subs(x, oo) == AccumBounds(-2, 2)
+
+def test_subs_undefined_result():
+    x, y = symbols("x, y")
+    expr = x/y
+    assert expr.subs([(x, 0), (y, 0)]) == 0
+    assert expr.subs([(x, 0), (y, 0)], simultaneous=True) == nan
```

Comment:
@oscarbenjamin , I found `simultaneous` a good option for evaluating some of the undefined expression and there is not any specific test case which exactly shows its purpose.